internal/build/lifecycle_execution.go

```diff
@@ -328,6 +328,10 @@ func (l *LifecycleExecution) Create(ctx context.Context, buildCache, launchCache
 		flags = append(flags, "-uid", strconv.Itoa(l.opts.UID))
 	}
 
+	for _, reg := range l.opts.InsecureRegistries {
```

Comment:
I think we need to add a validation for platform API version, something like

```go
if l.platformAPI.AtLeast("0.13") {
  // we can add the `-insecure-registry` flag when calling the lifecycle.
}
```

Reply:
Do we need to add validation only for create or do we need to add for all like `analyze`, `restore`, `export`, and `rebase`?

Reply:
I think in all places where we are sending the values to lifecycle, basically because only those lifecycle will be able to handle the insecure registry

Reply:
Done!